python/lsst/afw/coord/coord.cc

```diff
@@ -0,0 +1,300 @@
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/operators.h>
+
+#include "lsst/afw/coord/Coord.h"
+
+namespace py = pybind11;
+
+using namespace lsst::afw::coord;
+
+PYBIND11_DECLARE_HOLDER_TYPE(MyCoordType, std::shared_ptr<MyCoordType>);
+
+PYBIND11_PLUGIN(_coord) {
+    py::module mod("_coord", "Python wrapper for afw _coord library");
+
+    /* Types and enums */
+    py::enum_<CoordSystem>(mod, "CoordSystem")
+        .value("UNKNOWN", CoordSystem::UNKNOWN)
+        .value("FK5", CoordSystem::FK5)
+        .value("ICRS", CoordSystem::ICRS)
+        .value("GALACTIC", CoordSystem::GALACTIC)
+        .value("ECLIPTIC", CoordSystem::ECLIPTIC)
+        .value("TOPOCENTRIC", CoordSystem::TOPOCENTRIC)
+    .export_values();
+
+    mod.def("makeCoordEnum", makeCoordEnum);
+
+    py::class_<Coord, std::shared_ptr<Coord>> clsCoord(mod, "Coord");
+
+    /* Constructors */
+    clsCoord.def(py::init<lsst::afw::geom::Point2D const, lsst::afw::geom::AngleUnit, double const>(),
+        py::arg("p2d"), py::arg("unit") = lsst::afw::geom::degrees, py::arg("epoch") = 2000.0);
+    clsCoord.def(py::init<lsst::afw::geom::Point3D const, double const, bool, lsst::afw::geom::Angle const>(),
+        py::arg("p2d"), py::arg("epoch") = 2000.0, py::arg("normalize") = true, py::arg("defaultLongitude") = lsst::afw::geom::Angle(0.));
```

Comment:
I can think of two possible ways around this:
 - Instead of specifying the default arguments in this way, we wrap each different version of the function (corresponding to different numbers of arguments provided) as a different overload.  That would be *very* verbose in our wrapper definition code, and depending on how pybind11 does its overload resolution, it could degrade the performance of the wrappers slightly.
 - We could (as a matter of convention) define static data members for all of our default values (or just any that are conceivably subject to change), and use those in both the wrapper definitions and the original function signatures.  I'm slightly worried about [static initialization order problems](https://isocpp.org/wiki/faq/ctors#static-init-order), but we can probably come up with a safe approach.